File: C1.UWP.FlexChart/VB/FinancialChartExplorer/EnumConverter.vb
Imports C1.Chart.Finance
Imports C1.Chart
Imports MovingAverageType = C1.Chart.MovingAverageType
Imports C1.Xaml.Chart.Interaction

Public Class EnumConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Return value.ToString()
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        If targetType Is GetType(FitType) Then
            Return CType([Enum].Parse(GetType(FitType), value.ToString()), FitType)
        ElseIf targetType Is GetType(MovingAverageType) Then
            Return CType([Enum].Parse(GetType(MovingAverageType), value.ToString()), MovingAverageType)
        ElseIf targetType Is GetType(RangeMode) Then
            Return CType([Enum].Parse(GetType(RangeMode), value.ToString()), RangeMode)
        ElseIf targetType Is GetType(DataFields) Then
            Return CType([Enum].Parse(GetType(DataFields), value.ToString()), DataFields)
        ElseIf targetType Is GetType(LabelPosition) Then
            Return CType([Enum].Parse(GetType(LabelPosition), value.ToString()), LabelPosition)
        ElseIf targetType Is GetType(LineMarkerAlignment) Then
            Return CType([Enum].Parse(GetType(LineMarkerAlignment), value.ToString()), LineMarkerAlignment)
        ElseIf targetType Is GetType(LineMarkerInteraction) Then
            Return CType([Enum].Parse(GetType(LineMarkerInteraction), value.ToString()), LineMarkerInteraction)
        ElseIf targetType Is GetType(LineMarkerLines) Then
            Return CType([Enum].Parse(GetType(LineMarkerLines), value.ToString()), LineMarkerLines)
        ElseIf targetType Is GetType(Boolean) Then
            Return Boolean.Parse(value.ToString())
        End If
        Return value
    End Function
End Class

Public Class IntToVisibilityConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim index As Integer = Integer.Parse(value.ToString())
        Dim para As Integer = Integer.Parse(parameter.ToString())
        If index = para Then
            Return Visibility.Visible
        Else
            Return Visibility.Collapsed
        End If
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class

Public Class ReversedConverter
    Implements IValueConverter
    Public Function Convert(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.Convert
        Dim val As Boolean = Boolean.Parse(value.ToString())
        Return Not val
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, language As String) As Object Implements IValueConverter.ConvertBack
        Throw New NotImplementedException()
    End Function
End Class

File: C1.UWP.FlexChart/CS/FinancialChartExplorer/EnumConverter.cs
using System;
using Windows.UI.Xaml.Data;
using C1.Chart;
using C1.Chart.Finance;
using MovingAverageType = C1.Chart.MovingAverageType;
using C1.Xaml.Chart.Interaction;
using Windows.UI.Xaml;

namespace FinancialChartExplorer
{
    public class EnumConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return value.ToString();
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            if (targetType == typeof(FitType))
            {
                return (FitType)Enum.Parse(typeof(FitType), value.ToString());
            }
            else if (targetType == typeof(MovingAverageType))
            {
                return (MovingAverageType)Enum.Parse(typeof(MovingAverageType), value.ToString());
            }
            else if (targetType == typeof(RangeMode))
            {
                return (RangeMode)Enum.Parse(typeof(RangeMode), value.ToString());
            }
            else if (targetType == typeof(DataFields))
            {
                return (DataFields)Enum.Parse(typeof(DataFields), value.ToString());
            }
            else if (targetType == typeof(LabelPosition))
            {
                return (LabelPosition)Enum.Parse(typeof(LabelPosition), value.ToString());
            }
            else if (targetType == typeof(LineMarkerAlignment))
            {
                return (LineMarkerAlignment)Enum.Parse(typeof(LineMarkerAlignment), value.ToString());
            }
            else if (targetType == typeof(LineMarkerInteraction))
            {
                return (LineMarkerInteraction)Enum.Parse(typeof(LineMarkerInteraction), value.ToString());
            }
            else if (targetType == typeof(LineMarkerLines))
            {
                return (LineMarkerLines)Enum.Parse(typeof(LineMarkerLines), value.ToString());
            }
            else if (targetType == typeof(PointAndFigureScaling))
            {
                return (PointAndFigureScaling)Enum.Parse(typeof(PointAndFigureScaling), value.ToString());
            }
            else if (targetType == typeof(bool))
            {
                return bool.Parse(value.ToString());
            }

            return value;
        }
    }

    public class IntToVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            int index = int.Parse(value.ToString());
            int para = int.Parse(parameter.ToString());
            return index == para ? Visibility.Visible : Visibility.Collapsed;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }

    public class ReversedConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            var val = bool.Parse(value.ToString());
            return !val;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }
}
